openstack/compute/v2/extensions/keypairs/requests.go

```diff
@@ -81,9 +85,85 @@ func Get(client *gophercloud.ServiceClient, name string) (r GetResult) {
 	return
 }
 
+// GetOptsBuilder allows extensions to add additional parameters to the
+// Get request.
+type GetOptsBuilder interface {
+	ToKeyPairGetQuery() (string, error)
+}
+
+// GetOpts enables retrieving KeyPairs based on specific attributes.
+type GetOpts struct {
+	// UserID is the user ID that owns the key pair.
+	// This requires microversion 2.10 or higher.
+	UserID string `q:"user_id"`
+}
+
+// ToKeyPairGetQuery formats a GetOpts into a query string.
+func (opts GetOpts) ToKeyPairGetQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// GetWithOpts is similar to Get, but can take a GetOpts struct.
+// The options within GetOpts require the client to have a microversion set.
+// The original Get function is retained for non-microverisoned requests.
+func GetWithOpts(client *gophercloud.ServiceClient, name string, opts GetOptsBuilder) (r GetResult) {
```

Comment:
@jtopjian `GetWithOpts` func looks hacky. I'd extend the `Get` function, even if it breaks the previous API.